tests/kafkatest/services/kafka/kafka.py

```diff
@@ -126,9 +126,14 @@ def set_version(self, version):
 
     @property
     def security_config(self):
-        return SecurityConfig(self.context, self.security_protocol, self.interbroker_security_protocol,
+        config = SecurityConfig(self.context, self.security_protocol, self.interbroker_security_protocol,
                               zk_sasl=self.zk.zk_sasl,
                               client_sasl_mechanism=self.client_sasl_mechanism, interbroker_sasl_mechanism=self.interbroker_sasl_mechanism)
+        for protocol in self.port_mappings:
+            port = self.port_mappings[protocol]
+            if port.open:
+                config.enable_security_protocol(port.name)
+        return config
```

Comment:
@ijuma Yes, that was my first inclination too. And then I thought there are so many parameters in the constructor already and you would only ever want to do this on the broker-side. So went for a method instead. Happy to change though if it is preferable to add on the constructor.